algoliasearch/check_settings.go

```diff
@@ -0,0 +1,72 @@
+package algoliasearch
+
+import "fmt"
+
+func checkSettings(settings map[string]interface{}) error {
+	for k, v := range settings {
+		switch v.(type) {
+		case []Alternative:
+			if k != "altCorrections" {
+				return fmt.Errorf("SettingsMap field %s is invalid", k)
```

Comment:
Same for pretty much all Errorf: You can use `invalidParameter` in `utils.go` of my PR